### PR TITLE
Integrate genome features component

### PR DESCRIPTION
**Pull request by nathandunn**
Would fix this, using the npm component: 

https://agr-jira.atlassian.net/browse/AGR-1146

- [x] remove old code
- [x] get working with proper npm versions
- [ ] add overflow properly

### CODE REVIEW

**Comment by nathandunn** [comment]
Going to close this for now.  I think there are a few things in the widget that aren't quite rendering correctly.     See https://github.com/GMOD/GenomeFeatureComponent/issues/10

**Comment by nathandunn** [comment]
Using 0.2.2 and getting this:

```
Uncaught (in promise) TypeError: display_feats.indexOf is not a function
    at eval (RenderFunctions.js:112)
    at Array.forEach (<anonymous>)
    at findRange (RenderFunctions.js:111)
    at IsoformTrack.DrawTrack (IsoformTrack.js:46)
    at eval (IsoformTrack.js:260)
```

**Comment by nathandunn** [comment]
@sibyl229  / @paaatrick / @pjhale  thought you might be able to advise on best how to do this:

Works, except that it seems to have trouble pulling in the proper CSS: 

screen shot 2018-12-19 at 1 45 05 pm

I have this working in vuejs using SCSS on the newer monarch-ui:

https://github.com/monarch-initiative/monarch-ui/blob/master/src/components/GenomeFeature.vue#L120

The GenomeFeatureComponent is VanillaJS + D3 (https://github.com/GMOD/GenomeFeatureComponent) and I can add whatever I want to it. 

Also happy to split up the CSS if need be.  I'd prefer not to have a default CSS.  The demo injects the CSS in a similar way:

https://github.com/GMOD/GenomeFeatureComponent/blob/development/src/demo/index.ejs#L5

**Comment by nathandunn** [comment]
And this isn't going anywhere, so after the holidays is fine as well, if you have any good ideas.

**Comment by sibyl229** [comment]
Hi @nathandunn I suspect the problem is due the AGR UI handles loading of CSS using CSS modules. Essentially, as the CSS is imported into JS (your react component), the css selectors (like classes) declared in your css files will be rewritten into a different string to avoid global namespace collision. 

If I suspect correct, here is what you need to do to fix: https://github.com/gajus/babel-plugin-react-css-modules#css-modules

File: src/components/expression/expressionComparisonRibbon.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ class ExpressionComparisonRibbon extends React.Component {
                     geneId={geneId}
                     groups={groups}
                     onClick={this.handleBlockClick}
-                    overrideColor={geneTaxon === TAXON_IDS.HUMAN && '#dedede'}
+                    overrideColor={geneTaxon === TAXON_IDS.HUMAN && '#dedede' ? 't' : 'f'}
```
Comment (paaatrick) [inline src/components/expression/expressionComparisonRibbon.js]: This isn't the right way to fix the proptypes warning and would actually introduce a bug in the expression ribbon. Please remove.

**Comment by paaatrick** [comment]
Sibyl's comment is correct if you need to add custom styling that exists only in the Alliance code.

If the `genomefeaturecomponent` package distributes its own CSS then you might just need to add it similarly to how other libraries' CSS gets added:
https://github.com/alliance-genome/agr_ui/blob/develop/src/index.js#L6

**Comment by nathandunn** [comment]
Awesome.  Thanks for the pointers.  Will fix when I get back from break and resubmit without the expression “fix”.

Cheers,

Nathan

**Comment by nathandunn** [comment]
Works when it runs (npm start), but now I get this on the tests:

```
 WEBPACK  Compiled successfully in 3503ms

 MOCHA  Testing...

 RUNTIME EXCEPTION  Exception occurred while loading your tests

ReferenceError: window is not defined
    at Object.<anonymous> (/Users/nathandunn/repositories/agr_ui/node_modules/genomefeaturecomponent/dist/index.js:10:4)
    at Module._compile (module.js:653:30)
    at Module._extensions..js (module.js:664:10)
    at Object._module2.default._extensions.(anonymous function) [as .js] (/Users/nathandunn/repositories/agr_ui/node_modules/mocha-webpack/lib/util/registerRequireHook.js:141:62)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.genomefeaturecomponent (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/external "genomefeaturecomponent":1:1)
    at __webpack_require__ (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/webpack/bootstrap:19:1)
    at Module.<anonymous> (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/main.js:9925:80)
    at Module../src/containers/genePage/genomeFeatureWrapper.js (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/main.js:10161:30)
    at __webpack_require__ (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/webpack/bootstrap:19:1)
    at Module.<anonymous> (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/main.js:10194:80)
    at Module../src/containers/genePage/index.js (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/main.js:10497:30)
    at __webpack_require__ (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/webpack/bootstrap:19:1)
    at Module../src/containers/genePage/test.js (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/main.js:10767:64)
    at __webpack_require__ (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/webpack/bootstrap:19:1)
    at run (/Users/nathandunn/repositories/agr_ui/node_modules/mocha-webpack/lib/entry.js:3:1)
    at Array.forEach (<anonymous>)
    at Object../node_modules/mocha-webpack/lib/entry.js (/Users/nathandunn/repositories/agr_ui/node_modules/mocha-webpack/lib/entry.js:10:1)
    at __webpack_require__ (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/webpack/bootstrap:19:1)
    at ./node_modules/mocha-webpack/lib/entry.js.ids (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/webpack/bootstrap:83:1)
    at Object.<anonymous> (/Users/nathandunn/repositories/agr_ui/.tmp/mocha-webpack/1546461243902/main.js:87:10)
    at Module._compile (module.js:653:30)
    at Object._module2.default._extensions.(anonymous function) [as .js] (/Users/nathandunn/repositories/agr_ui/node_modules/mocha-webpack/lib/util/registerRequireHook.js:148:12)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at /Users/nathandunn/repositories/agr_ui/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/Users/nathandunn/repositories/agr_ui/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/Users/nathandunn/repositories/agr_ui/node_modules/mocha/lib/mocha.js:576:10)
    at /Users/nathandunn/repositories/agr_ui/node_modules/mocha-webpack/lib/runner/TestRunner.js:191:27
    at /Users/nathandunn/repositories/agr_ui/node_modules/mocha-webpack/lib/webpack/compiler/registerReadyCallback.js:26:7
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/nathandunn/repositories/agr_ui/node_modules/tapable/lib/HookCodeFactory.js:32:10), <anonymous>:27:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/nathandunn/repositories/agr_ui/node_modules/tapable/lib/Hook.js:154:20)
    at emitRecords.err (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:257:22)
    at Compiler.emitRecords (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:371:39)
    at emitAssets.err (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:251:10)
    at hooks.afterEmit.callAsync.err (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:357:14)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/nathandunn/repositories/agr_ui/node_modules/tapable/lib/HookCodeFactory.js:32:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/nathandunn/repositories/agr_ui/node_modules/tapable/lib/Hook.js:154:20)
    at asyncLib.forEach.err (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:354:27)
    at /Users/nathandunn/repositories/agr_ui/node_modules/neo-async/async.js:2825:7
    at done (/Users/nathandunn/repositories/agr_ui/node_modules/neo-async/async.js:2861:11)
    at /Users/nathandunn/repositories/agr_ui/node_modules/neo-async/async.js:2813:7
    at MemoryFileSystem.writeFile (/Users/nathandunn/repositories/agr_ui/node_modules/memory-fs/lib/MemoryFileSystem.js:328:9)
    at writeOut (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:338:29)
    at asyncLib.forEach (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:348:7)
    at baseEach (/Users/nathandunn/repositories/agr_ui/node_modules/neo-async/async.js:2416:9)
    at Object.each (/Users/nathandunn/repositories/agr_ui/node_modules/neo-async/async.js:2850:9)
    at emitFiles (/Users/nathandunn/repositories/agr_ui/node_modules/webpack/lib/Compiler.js:311:13)
    at Immediate.<anonymous> (/Users/nathandunn/repositories/agr_ui/node_modules/memory-fs/lib/MemoryFileSystem.js:288:4)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)

npm ERR! Test failed.  See above for more details.
```

**Comment by nathandunn** [comment]
So, the problem (not sure what the fix is) is in GenomeFeatureComponents properly exporting:

From `agr_ui/node_modules/genomefeaturecomponent/dist/index.js`

```
(function webpackUniversalModuleDefinition(root, factory) {
    if(typeof exports === 'object' && typeof module === 'object')
        module.exports = factory();
    else if(typeof define === 'function' && define.amd)
        define([], factory);
    else if(typeof exports === 'object')
        exports["GenomeFeatureViewer"] = factory();
    else
        root["GenomeFeatureViewer"] = factory();
})(window, function() {
```

and should be

```
(function webpackUniversalModuleDefinition(root, factory) {
    if(typeof exports === 'object' && typeof module === 'object')
        module.exports = factory();
    else if(typeof define === 'function' && define.amd)
        define([], factory);
    else if(typeof exports === 'object')
        exports["GenomeFeatureViewer"] = factory();
    else
        root["GenomeFeatureViewer"] = factory();
})(typeof self !== 'undefined' ? self : this, function() {
```

**Comment by nathandunn** [comment]
from https://medium.com/@JakeXiao/window-is-undefined-in-umd-library-output-for-webpack4-858af1b881df

**Comment by nathandunn** [comment]
@paaatrick  / @sibyl229  this should be ready for review .  Tests should pass now.

File: src/containers/genePage/genomeFeatureWrapper.js [hunk 2]
```diff
@@ -70,39 +67,41 @@ class GenomeFeatureViewer extends Component {
   }
 
 
-  componentDidMount() {
-    this.loadData();
-  }
-
   componentDidUpdate() {
```
Comment (sibyl229) [inline src/containers/genePage/genomeFeatureWrapper.js]: Hi @nathandunn not necessarily a problem, just a reminder that `componentDidUpdate` isn't called when the component initially mounts. _If_ you need to call `loadGenomeFeature` when the component initially mounts, you will need to call it in `componentDidMount`

**Pull request merged**